chromium_src/base/feature_override.h

```diff
@@ -0,0 +1,52 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_BASE_FEATURE_OVERRIDE_H_
+#define BRAVE_CHROMIUM_SRC_BASE_FEATURE_OVERRIDE_H_
+
+#include "base/feature_list.h"
+
+// Helpers to override base::Feature::default_state without patching.
+// Usage:
+// 1. Create chromium_src/.../features.cc override for a file that contains the
+//    feature to override.
+// 2. #include "base/feature_override.h"
+// 3. Declare ENABLE_FEATURE_BY_DEFAULT(...) or DISABLE_FEATURE_BY_DEFAULT(...).
+
+namespace base {
+namespace internal {
+
+class BASE_EXPORT FeatureDefaultStateOverride {
+ public:
+  FeatureDefaultStateOverride(const base::Feature& feature,
+                              FeatureState default_state);
+};
+
+}  // namespace internal
+}  // namespace base
+
+// clang-format off
+#define OVERRIDE_FEATURE_DEFAULT_STATE_UNIQUE(feature, default_state, key) \
+  _Pragma("clang diagnostic push")                                         \
+  _Pragma("clang diagnostic ignored \"-Wglobal-constructors\"")            \
+  static const ::base::internal::FeatureDefaultStateOverride               \
+      g_feature_default_state_override_##key(feature, default_state);      \
+  _Pragma("clang diagnostic pop")                                          \
+  static_assert(true, "") /* for a semicolon requirement */
```

Comment:
I saw some references to this pragmas in https://github.com/brave/brave-browser/issues/18829#issuecomment-950857801, but that's because I navigated to the issue from this PR to search for more context, otherwise this would be very confusing. Thus, I think it would be great if we could get some comments around this block of code explaining why it's like that.